packages/lesswrong/components/sunshineDashboard/ContentRejectionDialog.tsx

```diff
@@ -0,0 +1,116 @@
+import Button from '@material-ui/core/Button';
+import Checkbox from '@material-ui/core/Checkbox';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+import classNames from 'classnames';
+import React, { ChangeEventHandler, useState } from 'react';
+import { useMulti } from '../../lib/crud/withMulti';
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  dialogContent: {
+    width: 400,
+    backgroundColor: theme.palette.panelBackground.default,
+    padding: 12,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+  },
+  rejectionCheckboxes: {
+    display: 'flex',
+    flexDirection: 'column'
+  },
+  checkbox: {
+    paddingTop: 6,
+    paddingBottom: 6
+  },
+  modalTextField: {
+    marginTop: 10,
+  },
+  hideModalTextField: {
+    display: 'none'
+  }
+});
+
+const ContentRejectionDialog = ({classes, rejectContent}: {
+  classes: ClassesType,
+  rejectContent: (reason: string) => void,
+}) => {
+  const { LWTooltip } = Components;
+
+  const [selections, setSelections] = useState<Record<string,boolean>>({});
+  const [hideTextField, setHideTextField] = useState(true);
+  const [rejectedReason, setRejectedReason] = useState('');
+
+  const { results } = useMulti({
+    collectionName: 'ModerationTemplates',
+    terms: { view: 'rejectionModerationTemplates' },
+    fragmentName: 'ModerationTemplateFragment'
+  });
+
+  if (!results) return null;
+  
+  const rejectionReasons = Object.fromEntries(results.map(({name, contents}) => [name, contents?.html]))
+
+  const handleClick = () => {
+    rejectContent(rejectedReason);
+  };
+
+  const composeRejectedReason = (label: string, checked: boolean) => {
```

Comment:
This function confused me for a moment, had to realize that it iteratively builds up and is stateful.

Hmm, is it bad that it's stateful in this way?